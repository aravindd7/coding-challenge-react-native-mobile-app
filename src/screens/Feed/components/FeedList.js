import React from "react";
import { FlatList, RefreshControl } from "react-native";

import FeedItem from "./FeedItem";

export default ({ loading, feed, feed_data, feedUpdate }) => (
  <FlatList
    data={feed_data}
    keyExtractor={item => item.data.id}
    renderItem={({ item }) => <FeedItem item={item} />}
    refreshing={loading}
    refreshControl={
      <RefreshControl refreshing={loading} onRefresh={feedUpdate} />
    }
  />
);
