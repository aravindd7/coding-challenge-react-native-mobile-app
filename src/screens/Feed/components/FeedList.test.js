import React from "react";

import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow, render } from "enzyme";

import FeedList from "./FeedList";

configure({ adapter: new Adapter() });

const props = {
  loading: true,
  section: "top",
  feed_data: [
    {
      id: "1234",
      title: "title",
      author: "author",
      num_comments: 3,
      score: 2,
      date: 1517096810066,
      thumbnail: "thumbnail.jpg",
      url: "http://example.com",
      selectItem: () => {}
    }
  ],
  feedUpdate: () => {},
  selectItem: () => {},
  feedSectionChange: () => {}
};

it("match on Shallow mode", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<FeedList {...props} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("Full render", () => {
  const response = {
    data: {
      children: [
        {
          data: {
            id: "1234",
            title: "title",
            author: "author",
            num_comments: 3,
            score: 2,
            created_utc: 1517115254,
            thumbnail: "thumbnail.jpg",
            url: "http://example.com"
          }
        },
        {
          data: {
            id: "1234",
            title: "title",
            author: "author",
            num_comments: 3,
            score: 2,
            created_utc: 1517115254,
            thumbnail: "",
            url: "http://example.com",
            media: { oembed: { thumbnail_url: "thumbnail_url.jpg" } }
          }
        }
      ]
    }
  };

  window.fetch = jest
    .fn()
    .mockImplementation(() => Promise.resolve({ json: () => response }));

  const feedSectionChangeMock = jest.fn();

  const wrapper = shallow(<FeedList {...props} />);

// TODO
  // expect(wrapper.render()).toMatchSnapshot();
});
