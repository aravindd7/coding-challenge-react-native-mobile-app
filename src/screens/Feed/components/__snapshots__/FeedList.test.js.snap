// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match on Shallow mode 1`] = `
<FlatList
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "author": "author",
        "date": 1517096810066,
        "id": "1234",
        "num_comments": 3,
        "score": 2,
        "selectItem": [Function],
        "thumbnail": "thumbnail.jpg",
        "title": "title",
        "url": "http://example.com",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={true}
    />
  }
  refreshing={true}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
